internal/update/update.go

```diff
@@ -31,7 +33,48 @@ type StateEntry struct {
 	LatestRelease      ReleaseInfo `yaml:"latest_release"`
 }
 
-// CheckForUpdate checks whether this software has had a newer release on GitHub
+// ShouldCheckForUpdate decides whether we check for updates for the GitHub CLI and/or GitHub CLI extensions based on user preferences and current execution context.
+func ShouldCheckForUpdate() bool {
+	if os.Getenv("GH_NO_UPDATE_NOTIFIER") != "" {
```

Comment:
Talking with @williammartin, we recognize that some users will not want update notices about extensions despite our concerns with security or desires to support extension authors with newer releases.  **To that end, we will use a separate environment variable to allow those users to disable all extension update notices for now until we have a more robust mechanism to control extension-specific notices.**

To that end, we'll duplicate this function as its likely that the logic will continue to evolve separate for core `gh` versus extensions 👍